### PR TITLE
Slurm upgrade 16.08

### PR DESCRIPTION
**Pull request by martbhell**
With this update the slurm role will pause ( only ) on the slurmdbd host if the slurm version in 
<pre>
yum list installed slurm | grep slurm | awk '{print $2}' | cut -d'-' -f1|cut -d "." -f1-2|sed -e 's/\.//'
</pre>

and 

<pre>
{{ fgci_slurmrepo_version | replace('fgcislurm', '')}}
</pre>

are different (and if the first is not an empty string)

Some comments on this would be nice. What do you think @jabl  and @A1ve5 ?

Should we add notes in ansible for where users should look for documentation? FGI/FGCI confluence is not ideal as that's behind HAKA authentication

I mentioned in #63 that the cgroup variable default has changed. Fortunately we have already the new default set in this role, so we don't need to make SLURM version dependent variable files.

### CODE REVIEW

**Comment by A1ve5** [comment]
Shall we add a "just in case" DB backup if version changes and before pause?

**Comment by jabl** [comment]
Yes, might be a good idea to backup the DB first just in case. Additionally, maybe it would make sense to install a cron job to do periodic backups of the database regardless of whether we're updating it or not.

FWIW, at the moment our DB is 16 GB, so we need to be careful were we put the backups so we don't fill some unlucky partition.

**Comment by martbhell** [comment]
Ok. Do we need stop slurmdbd and slurmctld before dumping ( we don't for taito's daily dump) in this case? If we had a periodic dump working then that would ensure we have a good partition for this dump too and one could just run the dumpscript.

**Comment by martbhell** [comment]
tldr; Perhaps I am overcomplicating this? But after writing the below I'm leaning towards just templating in the dump-script and adding a "run the mysql dump" as part of the instructions.

---
This is the dump-script we use in taito: https://gist.github.com/felipelavinz/5797286 (it has the correct path for EL7 mysqldump).
One could use {{ ansible_mounts }} to assert (fail the ansible-play) that the path we are setting as the output directory has enough free space and also gathering the size of /var/lib/mysql/ibdata* files.
Maybe a with_nested after somehow (not sure if possible with jinja) splitting the path into [ /var/lib/mysql/dump, /var/lib/mysql, /var/lib, /var, / ] and then testing each of them against the ansible_mounts.

There is also the mysql_db that one can use like this to dump all databases like this:
<pre>
mysql_db: state=dump name=all target=/tmp/{{ inventory_hostname }}.sql
</pre>

We'd have to install ansible on the install node to run that inside a cron script.

Proposal for adding these tasks:
  - template in the dump script (copy of the gist above stored in ansible-role-slurm)

I'm proposing that we do not add these:
 - add yet another task file for this role called dbdump.yml
  - set and create default output directory for the dump script to /var/lib/mysql/backup
  - add a nightly cronjob
  - assert that there is enough free space in the mount for the backup at the output directory (and fail the play if there isn't)
 - in the upgrade.yml before the pause run the dumpscript
